### PR TITLE
 Change ValueErrors to InvalidMetadataErrors

### LINKED ISSUE #1251: Change ValueErrors to InvalidMetadataErrors
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 1.0.0.dev
* Python version: Any
* Operating System: Any

### Error Description

The validation methods in the [metadata.single_table](https://github.com/sdv-dev/SDV/blob/d28ec1496202efbac0c0747798464b85d7a1828d/sdv/metadata/single_table.py) module raise `ValueError`s, but the should raise `InvalidMetadataError`s.

### PR DESCRIPTION
**Pull request by frances-h**
Change `ValueError`s in  single_ and multi_table validation methods to `InvalidMetadataError`s

Resolve #1251

### CODE REVIEW

File: tests/unit/metadata/test_single_table.py [hunk 1]
```diff
@@ -784,14 +788,15 @@ def test_detect_from_dataframe(self, mock_log):
     def test_detect_from_csv_raises_value_error(self):
```
Comment (fealho) [inline tests/unit/metadata/test_single_table.py:788]: same

File: tests/unit/metadata/test_single_table.py [hunk 2]
```diff
@@ -713,14 +716,15 @@ def test_upate_column_no_sdtype(self, mock__validate_column_exists, mock__valida
     def test_detect_from_dataframe_raises_value_error(self):
```
Comment (fealho) [inline tests/unit/metadata/test_single_table.py:716]: Maybe change the `value_error` to just `error`, or something like that.

File: tests/unit/metadata/test_multi_table.py [hunk 3]
```diff
@@ -1718,7 +1721,7 @@ def test_load_from_json_path_does_not_exist(self, mock_path):
             - String representing a filepath.
 
         Side Effects:
-            - A ``ValueError`` is raised pointing that the ``file`` does not exist.
+            - An ``InvalidMetadataError`` is raised pointing that the ``file`` does not exist.
```
Comment (fealho) [inline tests/unit/metadata/test_multi_table.py:1724]: This test raises a ValueError, so you shouldn't update the docstring.

File: tests/unit/metadata/test_single_table.py [hunk 4]
```diff
@@ -1541,7 +1548,7 @@ def test_load_from_json_path_does_not_exist(self, mock_path):
             - String representing a filepath.
 
         Side Effects:
-            - A ``ValueError`` is raised pointing that the ``file`` does not exist.
+            - An ``InvalidMetadataError`` is raised pointing that the ``file`` does not exist.
```
Comment (fealho) [inline tests/unit/metadata/test_single_table.py:1551]: same

File: tests/unit/metadata/test_single_table.py [hunk 5]
```diff
@@ -1646,15 +1653,15 @@ def test_save_to_json_file_exists(self, mock_path):
...
+        raises an ``InvalidMetadataError``.
 
         Setup:
             - instance of ``SingleTableMetadata``.
         Mock:
             - Mock ``Path`` in order to point that the file does exist.
 
         Side Effects:
-            - Raise ``ValueError`` pointing that the file does exist.
+            - Raise ``InvalidMetadataError`` pointing that the file does exist.
```
Comment (fealho) [inline tests/unit/metadata/test_single_table.py:1664]: same

File: tests/unit/metadata/test_multi_table.py [hunk 6]
```diff
@@ -1788,15 +1791,15 @@ def test_save_to_json_file_exists(self, mock_path):
...
+        raises an ``InvalidMetadataError``.
 
         Setup:
             - instance of ``MultiTableMetadata``.
         Mock:
             - Mock ``Path`` in order to point that the file does exist.
 
         Side Effects:
-            - Raise ``ValueError`` pointing that the file does exist.
+            - Raise ``InvalidMetadataError`` pointing that the file does exist.
```
Comment (fealho) [inline tests/unit/metadata/test_multi_table.py:1802]: same

File: sdv/metadata/multi_table.py [hunk 7]
```diff
@@ -473,15 +480,15 @@ def add_table(self, table_name):
                 The name of the table to add to the metadata.
 
         Raises:
-            Raises ``ValueError`` if ``table_name`` is not valid.
+            Raises ``InvalidMetadataError`` if ``table_name`` is not valid.
         """
         if not isinstance(table_name, str) or table_name == '':
-            raise ValueError(
+            raise InvalidMetadataError(
```
Comment (fealho) [inline sdv/metadata/multi_table.py:486]: @amontanez24 The issue only mentioned the `validate` methods. I'm not sure if these should be changed as well?

File: sdv/metadata/multi_table.py [hunk 8]
```diff
@@ -597,7 +604,7 @@ def save_to_json(self, filepath):
                 String that represent the ``path`` to the ``json`` file to be written.
 
         Raises:
-            Raises a ``ValueError`` if the path already exists.
+            Raises a ``InvalidMetadataError`` if the path already exists.
```
Comment (fealho) [inline sdv/metadata/multi_table.py:607]: this raises a ValueError

File: sdv/metadata/multi_table.py [hunk 9]
```diff
@@ -673,7 +680,7 @@ def upgrade_metadata(cls, old_filepath, new_filepath):
                 String that represents the ``path`` to save the upgraded metadata to.
 
         Raises:
-            Raises a ``ValueError`` if the path already exists.
+            Raises a ``InvalidMetadataError`` if the path already exists.
```
Comment (fealho) [inline sdv/metadata/multi_table.py:683]: same

File: sdv/metadata/single_table.py [hunk 10]
```diff
@@ -166,22 +166,22 @@ def add_column(self, column_name, **kwargs):
...
+            - ``InvalidMetadataError`` if the column already exists.
+            - ``InvalidMetadataError`` if the ``kwargs`` do not contain ``sdtype``.
+            - ``InvalidMetadataError`` if the column has unexpected values or ``kwargs`` for the
+              given ``sdtype``.
             - ``InvalidMetadataError`` if the ``pii`` value is not ``True`` or ``False`` when
                present.
         """
         if column_name in self._columns:
-            raise ValueError(
+            raise InvalidMetadataError(
```
Comment (fealho) [inline sdv/metadata/single_table.py:177]: same for these, not sure if they should be updated.

**Review by fealho — CHANGES_REQUESTED** [review]

File: sdv/metadata/multi_table.py (hunk 7 above)
Comment (amontanez24) [reply sdv/metadata/multi_table.py:486]: Yeah we might as well. If it has to do with a bad input to the metadata I think we should change it for consistency

File: tests/unit/metadata/test_multi_table.py (hunk 3 above)
Comment (amontanez24) [reply tests/unit/metadata/test_multi_table.py:1724]: According to our new testing docstring guidelines I think you can actually just delete all the docstrings heere after line 1715

File: sdv/metadata/single_table.py [hunk 11]
```diff
@@ -539,14 +543,14 @@ def upgrade_metadata(cls, old_filepath, new_filepath):
                 String that represents the ``path`` to save the upgraded metadata to.
 
         Raises:
-            Raises a ``ValueError`` if the path already exists.
+            Raises a ``InvalidMetadataError`` if the path already exists.
```
Comment (amontanez24) [inline sdv/metadata/single_table.py:546]: This is also still a ValueError

**Review by amontanez24 — COMMENTED** [review]
As @fealho pointed out, there are a few errors that are related to file's not being found and should stay as `ValueError`s. Besides that I think this is good!

**Review by amontanez24 — APPROVED** [review]
Nice job! LGTM

**Review by fealho — APPROVED** [review]
Looks good, thanks for addressing the comments!

**Pull request merged**